src/Domain/Creators/Cylinder.hpp

```diff
@@ -101,16 +105,40 @@ class Cylinder : public DomainCreator<3> {
     static type default_value() noexcept { return false; }
   };
 
-  using options = tmpl::list<InnerRadius, OuterRadius, LowerBound, UpperBound,
-                             IsPeriodicInZ, InitialRefinement,
-                             InitialGridPoints, UseEquiangularMap>;
+  struct RadialPartitioning {
+    using type = std::vector<double>;
+    static type default_value() noexcept { return std::vector<double>{}; }
+    static constexpr OptionString help = {
+        "Radial coordinates of the boundaries splitting the outer shell "
+        "between InnerRadius and OuterRadius."};
+  };
+
+  struct HeightPartitioning {
+    using type = std::vector<double>;
+    static type default_value() noexcept { return std::vector<double>{}; }
+    static constexpr OptionString help = {
+        "z-coordinates of the boundaries splitting the domain into discs "
+        "between LowerBound and UpperBound."};
+  };
+
+  using options =
+      tmpl::list<InnerRadius, OuterRadius, LowerBound, UpperBound,
+                 IsPeriodicInZ, InitialRefinement, InitialGridPoints,
+                 UseEquiangularMap, RadialPartitioning, HeightPartitioning>;
 
   static constexpr OptionString help{
-      "Creates a 3D Cylinder with five Blocks.\n"
+      "Creates a right circular Cylinder with a square prism surrounded by \n"
+      "wedges. The wedges can be partitioned into shells and all blocks \n"
```

Comment:
Just so that the language is more precise, `The cylinder can be partitioned radially into multiple cylindrical shells as well as partitioned along the cylinder's height into multiple disks. Including this partitioning, the number of Blocks is given by...`